### PR TITLE
[VEN-2819] VIP to add markets, PSR, NativeTokenGateway and Prime token on OP mainnet

### PR DESCRIPTION
**Pull request by Debugger022**
## Description

Resolves VEN-2819

### CODE REVIEW

File: multisig/proposals/opmainnet/vip-003/index.ts [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    },
+    {
+      target: ACM,
+      signature: "giveCallPermission(address,string,address)",
+      params: [ZERO_ADDRESS, "setRewardTokenSpeeds(address[],uint256[],uint256[])", opmainnet.GUARDIAN],
+    },
+    {
+      target: ACM,
+      signature: "giveCallPermission(address,string,address)",
+      params: [ZERO_ADDRESS, "setLastRewardingBlock(address[],uint32[],uint32[])", opmainnet.GUARDIAN],
```
Comment (GitGuru7) [inline multisig/proposals/opmainnet/vip-003/index.ts:147]:
```suggestion
      params: [ZERO_ADDRESS, "setLastRewardingBlockTimestamps(address[],uint256[],uint256[])", opmainnet.GUARDIAN],
```

Comment (Debugger022) [reply multisig/proposals/opmainnet/vip-003/index.ts]: Fixed, 7280bcc18d462c253bf8d1ee55537a91da273e07

File: multisig/proposals/opmainnet/vip-003/index.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+
+export const VWBTC_CORE = "0x9EfdCfC2373f81D3DF24647B1c46e15268884c46";
+export const VWETH_CORE = "0x66d5AE25731Ce99D46770745385e662C8e0B4025";
+export const VUSDT_CORE = "0x37ac9731B0B02df54975cd0c7240e0977a051721";
+export const VOP_CORE = "0x6b846E3418455804C1920fA4CC7a31A51C659A2D";
+export const VUSDC_CORE = "0x1C9406ee95B7af55F005996947b19F91B6D55b15";
+
+const CHAINLINK_USDC_FEED = "0x16a9FA2FDa030272Ce99B29CF780dFA30361E0f3";
+const STALE_PERIOD_26H = 60 * 60 * 26; // 26 hours (pricefeeds with heartbeat of 24 hr)
+const USDCe = "0x7F5c764cBc14f9669B88837ca1490cCa17c31607";
```
Comment (GitGuru7) [inline multisig/proposals/opmainnet/vip-003/index.ts:24]: Shouldn't it be USDC 0x0b2c639c533813f4aa9d7837caf62653d097ff85 ?

**Review by GitGuru7 — APPROVED** [review]

File: multisig/simulations/opmainnet/vip-004/index.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  let accessControlManager: Contract;
+  let psrSigner: SignerWithAddress;
+
+  before(async () => {
+    protocolShareReserve = await ethers.getContractAt(PSR_ABI, PSR);
+    accessControlManager = await ethers.getContractAt(ACM_ABI, ACM);
+
+    psrSigner = await initMainnetUser(PSR, ethers.utils.parseEther("1"));
+    nativeTokenGateway = await ethers.getContractAt(GATEWAY_ABI, NATIVE_TOKEN_GATEWAY_CORE_POOL);
+    await pretendExecutingVip(await vip004());
```
[the remark this answers is not in the document]
Comment (chechu) [reply multisig/simulations/opmainnet/vip-004/index.ts:32]: 102f3aad97a858467d11b09c062046a49bdd5210

File: multisig/simulations/opmainnet/vip-005/index.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      await accounts[0].sendTransaction({ to: opmainnet.GENERIC_TEST_USER_ACCOUNT, value: parseUnits("1") });
+      await accounts[0].sendTransaction({ to: opmainnet.GUARDIAN, value: parseUnits("1") });
+
+      await network.provider.send("hardhat_setCode", [XVS_ADMIN, "0x"]);
+
+      await accounts[0].sendTransaction({ to: XVS_ADMIN, value: parseUnits("4") });
+      await xvs.mint(opmainnet.GENERIC_TEST_USER_ACCOUNT, parseUnits("1000", 18));
+
+      await pretendExecutingVip(await vip002());
+      await pretendExecutingVip(await vip003());
```
[the remark this answers is not in the document]
Comment (chechu) [reply multisig/simulations/opmainnet/vip-005/index.ts:56]: 102f3aad97a858467d11b09c062046a49bdd5210

File: multisig/simulations/opmainnet/vip-004/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (chechu) [reply multisig/simulations/opmainnet/vip-004/index.ts:32]: 102f3aad97a858467d11b09c062046a49bdd5210

File: multisig/simulations/opmainnet/vip-005/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (chechu) [reply multisig/simulations/opmainnet/vip-005/index.ts:56]: 102f3aad97a858467d11b09c062046a49bdd5210

**Review by chechu — COMMENTED** [review]
Commands LGTM

File: multisig/simulations/opmainnet/vip-003/index.ts [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+          const expectedSupply = parseUnits(params.initialSupply, 8);
+          const vToken = await ethers.getContractAt(VTOKEN_ABI, vTokens[symbol]);
+          expect(await vToken.balanceOf(params.vTokenReceiver)).to.equal(expectedSupply);
+        });
+      }
+    });
+
+    describe("Risk parameters", () => {
+      for (const [symbol, params] of Object.entries(riskParameters) as [VTokenSymbol, RiskParameters][]) {
+        describe(`${symbol} risk parameters`, () => {
```
[the remark this answers is not in the document]
Comment (Debugger022) [reply multisig/simulations/opmainnet/vip-003/index.ts]: Refactored, e4ee00934666e253ae1efde836ab639b2b887103

File: multisig/simulations/opmainnet/vip-003/index.ts [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+            {
+              base: model.base,
+              multiplier: model.multiplier,
+              jump: model.jump,
+              kink: model.kink,
+            },
+            BLOCKS_PER_YEAR,
+          );
+        }
+      }
```
[the remark this answers is not in the document]
Comment (Debugger022) [reply multisig/simulations/opmainnet/vip-003/index.ts:501]: Added, e4ee00934666e253ae1efde836ab639b2b887103

**Review by GitGuru7 — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/382?chainId=56

**Pull request merged**